### ISSUE TITLE
Invalid SQL generated when using nullable data types for Derby 

### ISSUE THREAD

**Issue by msteiger** [jOOQ/jOOQ#4321]
I'm getting invalid SQL for the following snippet:

```java
String url = "jdbc:derby:memory://localhost:1527/servers;create=true";
try (Connection conn = DriverManager.getConnection(url)) {
    DSL.using(conn)
        .createTable("t")
        .column("n", SQLDataType.VARCHAR)
        .execute();
}
```

The generated SQL statement is
```sql
create table "t"("n" varchar null)
```

Apache Derby doesn't accept the `null` part of it:

    ERROR 42X01: Syntax error: Encountered "null" at line 1, column 37.

It does work with `VARCHAR.nullable(false)` though:

```sql
create table "t"("n" varchar(32672) not null)
```

Also, the same snippet works fine when using H2 instead of Derby:
```sql
String url = "jdbc:h2:mem:test";
```

I'm using jOOQ at version 3.6.1 and Derby v10.11.1.1.

**Comment by lukaseder** [comment]
Interesting, this must have escaped our integration tests. Thanks for reporting!

As a workaround, I suspect you could replace `"(?<!not )null"` by `""` via an `ExecuteListener` in generated SQL

**Comment by msteiger** [comment]
Thanks for the quick fix - greatly appreciated!

**Comment by lukaseder** [comment]
Sure, no worries. Just noticed that we've currently deactivated integration tests for Derby DDL statements, as we hadn't fully supported them yet. We'll see that more fixes can make it into 3.6.2 (e.g. #4324)

**Comment by lukaseder** [comment]
`ALTER TABLE` is also affected
